crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -28,8 +28,11 @@ pub struct DynamicScene {
 
 /// A reflection-powered serializable representation of an entity and its components.
 pub struct DynamicEntity {
-    /// The transiently unique identifier of a corresponding `Entity`.
-    pub entity: u32,
+    /// The identifier of the entity, unique within a scene (and the world it may have been generated from).
+    ///
+    /// Components in the scene that reference other entities reference them through this identifier, and so this
+    /// identifier must be consistent with the references contained within components.
```

Comment:
Nit: This is still a bit confusing to me. Could it be simpler to state something like:
```suggestion
    /// Components that reference this entity must consistently use this identifier.
```

Or maybe I'm missing something?